### ISSUE TITLE
verbose = 0 still gives output?

### ISSUE THREAD

**Issue by beckermr** [rmjarvis/TreeCorr#27]
It appears that verbose = 0 still produces output to stdout or stderr. Is this the correct behavior? Is there anyway to make the code completely silent?

**Comment by rmjarvis** [comment]
I thought the only output you could get with verbose=0 was when there was an error that stops the program.  I.e. an exception that doesn't get caught by anyone.

What output are you seeing?  It's certainly possible that I missed something.

**Comment by beckermr** [comment]
I am seeing output that tells me when objects are created and destroyed and their addresses. I also see the variance in the shear field and bits of info like that.

**Comment by rmjarvis** [comment]
Sorry, I need more detail, since I can't reproduce this.  

What version of treecorr are you using?  How are you using it (via corr2 or within python)?  How are you setting verbose = 0?

And then precisely what is the output?

**Comment by beckermr** [comment]
Sure. Here is my config dict
```python
config = {}
config['do_auto_corr'] = False
config['min_sep'] = 1.0
config['max_sep'] = 400.0
config['nbins'] = 25
config['sep_units'] = 'arcmin'
config['verbose'] = 0
config['num_threads'] = nt
```
I am calling the code in python.

Here is the output I get
```
Start BuildGGCorr
corr = 0x122cf60
Start BuildGFieldSphere
Starting to Build Field with 3047164 objects
r = 0
Built celldata with 3047164 entries
Field has 7 top-level nodes.  Building lower nodes...
field = 0x18f6f90
Start ProcessCrossGGSphere
Start BuildGGCorr
corr = 0x1697d50
Start BuildGFieldSphere
Starting to Build Field with 3047164 objects
r = 0
Built celldata with 3047164 entries
Field has 7 top-level nodes.  Building lower nodes...
field = 0x1338d40
Start ProcessCrossGGSphere
Start BuildGGCorr
corr = 0x218a7160
Start ProcessCrossGGSphere
Start BuildKKCorr
corr = 0x150c2340
Start BuildKFieldSphere
Starting to Build Field with 3047164 objects
r = 0
Built celldata with 3047164 entries
Field has 7 top-level nodes.  Building lower nodes...
field = 0x1914020
Start BuildKFieldSphere
Starting to Build Field with 3047164 objects
r = 0
...
```

My treecorr version is 3.1.0.

**Comment by rmjarvis** [comment]
OK.  Found it.  Sorry about that.  It looks like I left the debugging stream turned on in the C++ layer, which is decoupled from the python-layer logging.

I'm not actually sure why I'm not getting the same thing on my system, but I can at least tell you how to fix it on yours, assuming you are compiling from a tarball or git clone, rather than just using pip.

Change line 21 of src/Field.cpp from
```
std::ostream* dbgout=&std::cerr;
```
to
```
std::ostream* dbgout=0
```
And rebuild.  That should turn off all this extra output.

I've fixed this on master, so it will make it into the next pip release as well.  I should probably apply this to the 3.1 branch as well and make 3.1.1 on pip.
